### ISSUE TITLE
Tilt crashed with a not http link

### ISSUE THREAD

**Issue by redvex** [tilt-dev/tilt#4945]
### Expected Behavior
When adding a not HTTP link in a resource, Tilt should show that link

### Current Behavior
The web app becomes unresponsive and it's hard to stop the process with CTRL+c.

### Steps to Reproduce

1. I've added a MySQL link in one of my resource:
```
dc_resource('mainapp',
            links=[
              link('mysql://root:password@localhost:3306/mydbname', 'Local DB')
            ],
            labels=['MainApp']
) 
```
2. TIlt try to rebuild the Tiltfile, but the process never completes
![Tilt not working](https://dsc.cloud/d91a06/Screenshot-2021-09-09-at-09.52.39.png "Tilt not working")

3. If I try to reload the page, it doesn't work
4. If I try to close the process on the terminal, it doesn't work immediately (It does after a couple of ctrl+c)
5. If I try to run `tilt up` again, the process start, but `(space)` doesn't open the terminal
6. If I change the URL from `mysql://` to `http://` and restart Tilt, it works, but that's not the link I need.

#### `tilt doctor` Output

```
$ tilt doctor
Tilt: v0.22.7, built 2021-09-03
System: darwin-amd64
---
Docker
- Host: [default]
- Server Version: 20.10.8
- API Version: 1.41
- Builder: 2
---
Kubernetes
- Env: docker-for-desktop
- Context: docker-desktop
- Cluster Name: docker-desktop
- Namespace: default
- Container Runtime: docker
- Version: v1.21.4
- Cluster Local Registry: none
---
Thanks for seeing the Tilt Doctor!
Please send the info above when filing bug reports. 💗

The info below helps us understand how you're using Tilt so we can improve,
but is not required to ask for help.
---
Analytics Settings
--> (These results reflect your personal opt in/out status and may be overridden by an `analytics_settings` call in your Tiltfile)
- User Mode: opt-in
- Machine: 89d34649937db2fa2f05bc53d5b0593b
- Repo: 1Gy8Wn6qHhxrZpKcmQlMsw==
```

#### About Your Use Case
Would be helpful for my team to have the db link on our resources to access the local (and staging DB) using a client (TablePlus or SequelPRO for example)

**Comment by milas** [comment]
Thanks for the report! A fix for this will be in the next Tilt release.

Once it's released and you upgrade, I'd recommend using [`version_settings`](https://docs.tilt.dev/api.html#api.version_settings) (or [`min_tilt_version`](https://github.com/tilt-dev/tilt-extensions/tree/master/min_tilt_version) extension) to avoid teammates on older Tilt versions from running into a hang when starting the project.

**Comment by milas** [comment]
This is now fixed in [v0.22.8+](https://github.com/tilt-dev/tilt/releases/latest) - if you still see issues after upgrading, feel free to comment here or open a new issue!
